Source/Core/RequestQueue.js

```diff
@@ -0,0 +1,186 @@
+                return this._length;
+            }
+        }
+    });
+
+    /**
+     * Get the request at the given index.
+     *
+     * @param {Number} index The index of the request.
+     *
+     * @return {Request} The request at the given index.
+     */
+    RequestQueue.prototype.get = function(index) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number.greaterThanOrEquals('index', index, 0);
+        Check.typeOf.number.lessThan('index', index, this._length);
+        //>>includeEnd('debug');
+        return this._array[index];
+    };
+
+    /**
+     * Insert a request into the queue. If the length would grow greater than the maximum length
+     * of the queue, the lowest priority request is removed and returned.
+     *
+     * @param {Request} request The request to insert.
+     *
+     * @return {Request|undefined} The request that was removed from the queue if the queue is at full capacity.
+     */
+    RequestQueue.prototype.insert = function(request) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('request', request);
+        //>>includeEnd('debug');
+
+        var array = this._array;
+        var previousLength = this._length;
+        var length = this._length;
+        var maximumLength = this._maximumLength;
+
+        if (length < maximumLength)
+        {
+            ++this._length;
+        }
+
+        if (previousLength === 0)
+        {
+            array[0] = request;
+            return;
+        }
+
+        var removedRequest;
+        var lastIndex = previousLength - 1;
+
+        if (previousLength === maximumLength) {
+            var lastRequest = array[lastIndex];
+            if (request.priority >= lastRequest.priority) {
+                // The array is full and the priority value of this request is too high to be inserted.
+                return request;
+            }
+            // The array is full and the inserted request pushes off the last request
+            removedRequest = lastRequest;
+            --lastIndex;
+        }
+
+        while (lastIndex >= 0 && request.priority < array[lastIndex].priority) {
+            array[lastIndex + 1] = array[lastIndex]; // Shift element to the right
+            --lastIndex;
+        }
+        array[lastIndex + 1] = request;
+
+        return removedRequest;
+    };
+
+    /**
+     * Call the given function for each request in the queue.
+     *
+     * @type {RequestQueue~ForEachCallback} The function to call.
+     */
+    RequestQueue.prototype.forEach = function(callback) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.func('callback', callback);
+        //>>includeEnd('debug');
+
+        var array = this._array;
+        var length = this._length;
+        for (var i = 0; i < length; ++i) {
+            callback(array[i]);
+        }
+    };
+
+    /**
+     * Sorts the queue.
+     */
+    RequestQueue.prototype.sort = function() {
```

Comment:
If we stay with this, is it worth extracting this out like we did for [mergeSort](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/mergeSort.js)?